lib/tf_lock/tf_lock_acquire.py

```diff
@@ -43,6 +46,31 @@ def tf_lock_acquire(root_module: Path) -> ExitCode:
                 )
                 return 0
 
+            else:
+                p = Parse(lock_user)
+                username = p.before.first("@")
+                # a pr holds the lock.
+                if "github" in str(lock_user):
+                    pr_number = p.after.first("@").before.first(".")
+                    repo_name = p.after.first(".").before.last(".", ".", ".")
+                    org_name = p.after.first(repo_name, ".").before.last(
+                        ".github"
+                    )
```

Comment:
can't you use the `TFLockUser.from_string()` method to do thisparsing? it's already being called on line 78 -- that can be moved up?